lib/puppet/rest/response_handler.rb

```diff
@@ -0,0 +1,33 @@
+module Puppet::Rest
+  module ResponseHandler
+
+    # Processe and return the content type and body of the response
+    # @param [Puppet::Rest::Response] response the HTTP response to process
+    # @return [String, String] the content type (minus its encoding) and
+    #         decompressed body of the response
+    def self.parse_response(response)
+      if response.content_type
+        [ response.content_type.gsub(/\s*;.*$/, ''), decompress_body(response) ]
+      else
+        raise _("No content type in http response; cannot parse")
```

Comment:
Is it a little aggressive to raise without a content type? Cursory browsing shows that it is not a [mandatory field](https://stackoverflow.com/questions/15860742/is-content-type-mandatory-in-http-post-request), but there might be other reasons I don't know about to raise in these situations?

Reply:
Not sure, I was maintaining our existing API here. I think we may use the value for something somewhere, and since I was eventually hoping to refactor our other HTTP stuff to use these utility methods, I decided to keep it.

Reply:
https://tools.ietf.org/html/rfc7231#section-3.1.1.5 indicates if the media type is unknown it should be treated as `application/octet-stream`. But I'm +1 to be more restrictive as it bad things will happen if we try to guess, e.g. trying to string concatenate binary and non-binary strings in ruby.